src/components/ha-cards.html

```diff
@@ -286,14 +291,20 @@
         });
       }
 
-      var sorted = window.HAWS.splitByGroups(states);
+      var splited = window.HAWS.splitByGroups(states);
+      if (orderedGroups) {
+        splited.groups.sort((gr1, gr2) => orderedGroups.indexOf(gr1.entity_id) -
```

Comment:
This is very inefficient and this calculation is done very often. Let's make `orderedGroups` passed in as an object mapping group entity_id to order

Something like this:

```json
{
  "group.kitchen": 2,
  "group.living_room": 1
}
```